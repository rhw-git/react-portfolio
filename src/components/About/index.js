import React from 'react';

import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Image from 'react-bootstrap/Image';

import profileImage from '../../assets/about/profile-img.png';

function About() {
  return (
    <Container fluid className="px-4" id="about">
      <Row>
        <Col xs={12} md={8}>
          <Row className="py-5 text-muted px-0">
            <Col xs={12}>
              <p>HELLO THERE!</p>
            </Col>
            <Col className="pb-4" xs={12} md={2}>
              <Image
                src={profileImage}
                roundedCircle
                style={{ width: '6rem' }}
              />
            </Col>
            <Col xs={12} md={10}>
              <h3 className="pb-5 text-sm-left text-md-middle pt-md-3">
                My name is Ruoha Wang, and I develop websites.
              </h3>
            </Col>
            <Col xs={12}>
              <h5 className="pb-3">
                I am based out of the Bay Area at Northern California.I work on
                both front-end & back-end applicaitons.
              </h5>
              <h5 className="pb-3">
                As a web Develop, I beleive that the good communications between
                all parties are essential for a successful project. So I am
                exciting to work with you closely to lauch your next site.{' '}
              </h5>
              <h5 className="pb-3">
                Let me know how can I be at your service, and I am just an email
                / phone call away.
              </h5>
            </Col>
          </Row>
        </Col>
        <Col xs={6} md={4}>
          <div className="py-5 text-muted">
            <p className="text-sm-right">EXPERIENCES WITH:</p>
            <div className="text-sm-right">
              <h5>CSS</h5>
              <h5>JavaScript</h5>
              <h5>Web / Third-Party / Server-Side APIs</h5>
              <h5>Node</h5>
              <h5>NoSQL</h5>
              <h5>PWA</h5>
            </div>
          </div>
        </Col>
      </Row>
    </Container>
  );
}

export default About;
