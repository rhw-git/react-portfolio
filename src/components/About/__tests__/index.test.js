// import all library and package
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
// import component
import About from '..';
// delete the component created for testing
afterEach(cleanup);
//---------------test-----------------//
describe('About component', () => {
  // renders About test
  it('render', () => {
    render(<About />);
  });
  // arrange
  const { asFragment } = render(<About />);
  // assert
  expect(asFragment()).toMatchSnapshot();
});
