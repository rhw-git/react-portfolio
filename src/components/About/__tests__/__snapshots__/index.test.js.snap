// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<DocumentFragment>
  <div
    class="px-4 container-fluid"
    id="about"
  >
    <div
      class="row"
    >
      <div
        class="col-md-8 col-12"
      >
        <div
          class="py-5 text-muted px-0 row"
        >
          <div
            class="col-12"
          >
            <p>
              HELLO THERE!
            </p>
          </div>
          <div
            class="pb-4 col-md-2 col-12"
          >
            <img
              class="rounded-circle"
              src="profile-img.png"
              style="width: 6rem;"
            />
          </div>
          <div
            class="col-md-10 col-12"
          >
            <h3
              class="pb-5 text-sm-left text-md-middle pt-md-3"
            >
              My name is Ruoha Wang, and I develop websites.
            </h3>
          </div>
          <div
            class="col-12"
          >
            <h5
              class="pb-3"
            >
              I am based out of the Bay Area at Northern California.I work on both front-end & back-end applicaitons.
            </h5>
            <h5
              class="pb-3"
            >
              As a web Develop, I beleive that the good communications between all parties are essential for a successful project. So I am exciting to work with you closely to lauch your next site. 
            </h5>
            <h5
              class="pb-3"
            >
              Let me know how can I be at your service, and I am just an email / phone call away.
            </h5>
          </div>
        </div>
      </div>
      <div
        class="col-md-4 col-6"
      >
        <div
          class="py-5 text-muted"
        >
          <p
            class="text-sm-right"
          >
            EXPERIENCES WITH:
          </p>
          <div
            class="text-sm-right"
          >
            <h5>
              CSS
            </h5>
            <h5>
              JavaScript
            </h5>
            <h5>
              Web / Third-Party / Server-Side APIs
            </h5>
            <h5>
              Node
            </h5>
            <h5>
              NoSQL
            </h5>
            <h5>
              PWA
            </h5>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
